core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -66,6 +66,11 @@ final class NonEmptyVector[+A] private (val toVector: Vector[A]) extends AnyVal
   def ++[AA >: A](other: Vector[AA]): NonEmptyVector[AA] = concat(other)
 
   /**
+    * Append this NEV to another NEV, producing a new `NonEmptyVector`.
+    */
+  def ++:[AA >: A](other: NonEmptyVector[AA]): NonEmptyVector[AA] = other.concatNev(this)
```

Comment:
Shouldn't these be the other way around? `++` for two `NEV`s  and `++:` for an `NEV` and a `Vector`?

Reply:
I'm not sure, I went with this https://github.com/typelevel/cats/pull/1838#issuecomment-334344911 - perhaps we should have both for both, since overloading is happening already anyway?

Reply:
This is to mimic the API in `Vector` and `List`.  Let's address #1838 first